### PR TITLE
Hw csv file

### PR DESCRIPTION
**Pull request by LizaBalshv**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+                int quantity = Integer.parseInt(parts[SIZE]);
+
+                if ("supply".equals(operationType)) {
+                    totalSupply += quantity;
+                } else if ("buy".equals(operationType)) {
+                    totalBuy += quantity;
+                }
+            }
+
+            writer.write("supply," + totalSupply + "\nbuy," + totalBuy + "\nresult,"
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:35]: use system.lineSeparator() instead

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+
+
 public class WorkWithFile {
+    private static final String SEPARATOR = "\\s*,\\s*";
+    private static final int TYPE = 0;
+    private static final int SIZE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:15]: To improve code structure and readability, consider breaking down the process into three distinct methods: readFile(), generateReport(), and writeToFile(). Each method will have a clear responsibility, enhancing maintainability. For instance:

1. Use readFile(fromFileName) to fetch and return the file's contents.
2. Pass the content to generateReport(data) to create the desired report.
3. Finally, call writeToFile(toFileName, report) to save the report.

Applying this structure to your current code would not only simplify the getStatistic method but also follow good coding practices. Here's a quick outline:

    String data = readFile(fromFileName);
    String report = generateReport(data);
    writeToFile(toFileName, report);

This approach will keep your code organized and make each part easier to test and debug.

**Review by okuzan — CHANGES_REQUESTED** [review]
Please make sure your build passes before creating a PR 🙏

**Comment by LizaBalshv** [comment]
> Please make sure your build passes before creating a PR 🙏

tell me please how I can find out whether my code passes the checklist before creation PR (find out whether all indents and spaces pass the checklist)?

**Review by okuzan — CHANGES_REQUESTED** [review]
To see if build passes locally you can use for example `mvn clean verify`, it should run checkstyle and tests

**Review by okuzan — APPROVED** [review]
Well done

**Pull request closed without merging**